src/Service/Indexer/IndexSchema2xDocument.php

```diff
@@ -0,0 +1,213 @@
+     */
+    public array $sp_citygov_phone;
+
+    /**
+     * @var string[]
+     */
+    public array $sp_citygov_email;
+
+    public ?string $sp_citygov_address;
+
+    public ?string $sp_citygov_startletter;
+
+    /**
+     * @var string[]
+     */
+    public array $sp_citygov_organisationtoken;
+
+    public int $sp_organisation;
+
+    public ?string $sp_citygov_firstname;
+
+    public ?string $sp_citygov_lastname;
+
+    /**
+     * List of Organisation Ids
+     * @var int[]
+     */
+    public array $sp_organisation_path;
+
+    /**
+     * List of Organisationnames
+     * @var string[]
+     */
+    public array $sp_citygov_organisation;
+
+    /**
+     * List of Productnames
+     * @var string[]
+     */
+    public array $sp_citygov_product;
+
+    public ?string $sp_citygov_function;
+    /**
+     * @var array<string,string|string[]>
+     */
+    private array $metaString = [];
+
+    /**
+     * @var array<string,string|string[]>
+     */
+    private array $metaText = [];
+
+    /**
+     * @var array<string,bool>
+     */
+    private array $metaBool = [];
+
+    /**
+     * @param string|string[] $value
+     */
+    public function setMetaString(string $name, string|array $value): void
+    {
+        $this->metaString[$name] = $value;
+    }
+
+    /**
+     * @param string|string[] $value
+     */
+    public function setMetaText(string $name, string|array $value): void
+    {
+        $this->metaText[$name] = $value;
+    }
+
+    public function setMetaBool(string $name, bool $value): void
+    {
+        $this->metaBool[$name] = $value;
+    }
+
+    /**
+     * @return array<String, mixed>
+     */
+    public function getFields(): array
+    {
+        $fields = get_object_vars($this);
+
+        // filter out inherited fields
+
+        $fields = array_filter(
+            $fields,
+            static function ($value, $key) {
+
+                if (is_null($value)) {
+                    return false;
+                }
+
+                $inheritedFields = [
+                    'fields',
+                    'modifiers',
+                    'fieldBoosts'
+                ];
+                if (in_array($key, $inheritedFields, true)) {
+                    return false;
+                }
+
+                if (
+                    $key === 'metaString' ||
+                    $key === 'metaText' ||
+                    $key === 'metaBool'
+                ) {
+                    return false;
+                }
+                return true;
+            },
+            ARRAY_FILTER_USE_BOTH
+        );
+        foreach ($this->metaString as $key => $value) {
+            $fields['sp_meta_string_' . $key] = $value;
+        }
+
+        foreach ($this->metaText as $key => $value) {
+            $fields['sp_meta_text_' . $key] = $value;
+        }
+
+        foreach ($this->metaBool as $key => $value) {
+            $fields['sp_meta_bool_' . $key] = $value;
+        }
+
+        return $fields;
+    }
```

Comment:
`$inheritedFields` and `$metaFields` should be constants. And if we define the table keys in `setMeta*` already like
```php
        $this->metaString['sp_meta_string_' . $key] = $value;
```
then we can do this:

```suggestion
    public function getFields(): array
    {
        return [
            ...array_filter(
                get_object_vars($this),
                fn($value, $key) => !is_null($value)
                      && !in_array($key, self::INHERITED_FIELDS, true)
                      && !in_array($key, self::META_FIELDS, true),
                ARRAY_FILTER_USE_BOTH
            ),
            ... $this->metaString,
            ... $this->metaText,
            ... $this->metaBool,
        ];
    }

```